common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1410,4 +1442,42 @@ long getPos() {
       return pos;
     }
   }
+
+  /**
+   * A class that wraps all the push-based metrics.
+   */
+  static class PushMergeMetrics implements MetricSet {
+    static final String NO_OPPORTUNITY_RESPONSES_METRIC = "couldNotFindOpportunityResponses";
+    static final String TOO_LATE_RESPONSES_METRIC = "tooLateResponses";
+    static final String PUSHED_BYTES_WRITTEN_METRIC = "pushedBytesWritten";
+    static final String CACHED_BLOCKS_BYTES_METRIC = "cachedBlocksBytes";
+
+    private final Map<String, Metric> allMetrics;
+    private final Meter noOpportunityResponses;
+    private final Meter tooLateResponses;
+    private final Meter pushedBytesWritten;
+    private final Counter cachedBlockBytes;
```

Comment:
We should document what these metrics mean?

Reply:
Fixed in the PR: https://github.com/apache/spark/pull/37638